### PR TITLE
Post comment edit delete add feat

### PR DESCRIPTION
**Pull request by samyakshah3008**
1. Create a Post
I can create a post with a "+" button or "Create new post" button which will open a textbox and will have a "Post" or "Submit" button.

2. Edit & Delete Post and comment

- Edit Post and comment
I can click on the "Edit Post" button which will open the same component with the textbox where we can edit a post & save it.

- Delete Post and comment
I can click on the "Delete Post" button which will delete a post.

### CODE REVIEW

File: src/components/Comments.jsx [hunk 1]
```diff
@@ -12,24 +12,34 @@ import {
...
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+  Textarea,
+  Input,
 } from "@chakra-ui/react";
```
Comment (Developers-codz) [inline src/components/Comments.jsx:25]: Appreciation -  for using Chakra UI

File: src/components/Comments.jsx [hunk 2]
```diff
@@ -12,24 +12,34 @@ import {
...
 import * as BsIcons from "react-icons/bs";
 
 import React from "react";
```
Comment (Developers-codz) [inline src/components/Comments.jsx:28]: importing react can be skipped as its no longer needed after version 17 also remove empty line

File: src/components/UserPost.jsx [hunk 3]
```diff
@@ -89,6 +105,18 @@ export default function UserPost({ post }) {
     onClose();
   };
 
+  const deletePostHandler = (post) => {
+    console.log(post, "FROM");
+    console.log(post._id, token);
```
Comment (Developers-codz) [inline src/components/UserPost.jsx:110]: remove console if no longer needed

File: src/components/UserPost.jsx [hunk 4]
```diff
@@ -89,6 +105,18 @@ export default function UserPost({ post }) {
...
+    dispatch(deletePost({ postId: post._id, token: token }));
+  };
+
+  const updatePostHandler = (post) => {
+    console.log(post, editPostInput);
```
Comment (Developers-codz) [inline src/components/UserPost.jsx:115]: here too

File: src/components/UserPost.jsx [hunk 5]
```diff
@@ -164,8 +192,53 @@ export default function UserPost({ post }) {
...
+                          <ModalHeader>Edit post</ModalHeader>
+
+                          <ModalCloseButton />
+                          <ModalBody>
+                            <Textarea
+                              _focus={{
+                                border: "2px",
+                                borderColor: "purple.600",
+                              }}
+                              // value={post.content}
```
Comment (Developers-codz) [inline src/components/UserPost.jsx]: remove commented code

File: src/store/postSlice.js [hunk 6]
```diff
@@ -107,23 +110,65 @@ export const deleteComments = createAsyncThunk(
...
+  "posts/editComments",
+  async ({ postId, commentId, commentData, token }) => {
+    try {
+      const response = await editCommentsServices(
+        postId,
+        commentId,
+        commentData,
+        token
+      );
+      console.log(response);
```
Comment (Developers-codz) [inline src/store/postSlice.js:123]: here

File: src/store/postSlice.js [hunk 7]
```diff
@@ -107,23 +110,65 @@ export const deleteComments = createAsyncThunk(
...
     }
   }
 );
 
+export const editPost = createAsyncThunk(
+  "/posts/editPosts",
+  async ({ id, content, token }) => {
+    try {
+      const response = await editPostService(id, content, token);
+      console.log(response);
```
Comment (Developers-codz) [inline src/store/postSlice.js:164]: here

File: src/store/postSlice.js [hunk 8]
```diff
@@ -160,6 +211,12 @@ const postSlice = createSlice({
     [deleteComments.fulfilled]: (state, action) => {
       state.posts = action.payload;
     },
+    [createNewPost.fulfilled]: (state, action) => {
+      state.posts = action.payload;
+    },
+    [editComments.fulfilled]: (state, action) => {
+      state.posts = action.payload;
```
Comment (Developers-codz) [inline src/store/postSlice.js:218]: suggestion -  handle rejected and pending states as well

**Review by Developers-codz — APPROVED** [review]
Good code quality and folder structure.
Added a few suggestions else its good to go 💯 💯

File: src/store/postSlice.js (hunk 8 above)
Comment (samyakshah3008) [reply src/store/postSlice.js:218]: Thank you for detailed review, Kajal!

**Comment by samyakshah3008** [comment]
Looks good to merge after one approval.

**Pull request merged**